File: Programming Basics/While Loop - Exercises/06.Cake.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int width = int.Parse(Console.ReadLine());
        int length = int.Parse(Console.ReadLine());

        int pieces = width * length;

        while (pieces >= 0)
        {
            string take = Console.ReadLine();

            if (take == "STOP")
            {
                Console.WriteLine($"{pieces} pieces are left.");
                System.Environment.Exit(0);
            }

            pieces -= int.Parse(take);
        }
        Console.WriteLine($"No more cake left! You need {Math.Abs(pieces)} pieces more.");
    }
}

File: Programming Basics - Python/While Loop - Exercises/06.Cake.py
width = int(input())
length = int(input())

total_pieces = width * length

pieces = input()

while pieces != "STOP":
    total_pieces -= int(pieces)

    if total_pieces < 0:
        print(f"No more cake left! You need {abs(total_pieces)} pieces more.")
        exit()

    pieces = input()

print(f"{total_pieces} pieces are left.")